curriculum/challenges/english/25-front-end-development/review-html-accessibility/671a87a39b88245a579c2271.md

```diff
@@ -9,7 +9,46 @@ dashedName: review-html-accessibility
+- **Large text or braille keyboards**: used by people with visual impairments to access the web.
+- **Screen magnifiers**: software programs that enlarge the content of a computer screen. They are used by people with low vision to access the web.
+- **Alternative pointing devices**: used by people with mobility impairments to access the web. This includes devices such as joysticks, trackballs, and touchpads.
+- **Voice recognition**: used by people with mobility impairments to access the web. It allows users to control a computer using their voice.
+
+## Accessibility Auditing Tools
+
+- **Google Lighthouse**: an open-source tool that can be used to audit the accessibility of a website. It provides a report with recommendations for improving accessibility.
+- **Wave**: a web accessibility evaluation tool that can be used to audit the accessibility of a website. It provides a report with recommendations for improving accessibility.
+- **IBM Equal Accessibility Checker**: a web accessibility evaluation tool that can be used to audit the accessibility of a website. It provides a report with recommendations for improving accessibility.
+
+## Accessibility Best Practices
+
+- **Proper heading level structure**: You should use proper heading levels to create a logical structure for your content. This helps screen readers and other assistive technologies understand the content of your website.
+- **Accessibility and Tables**: When using tables, you should use the `<th>` element to define header cells and the `<td>` element to define data cells. This helps screen readers and other assistive technologies understand the structure of the table.
+- **Importance for inputs to have an associated label**: You should use the `<label>` element to associate labels with form inputs. This helps screen readers and other assistive technologies understand the purpose of the input.
+- **Importance of good `alt` text**: You should use the `alt` attribute to provide a text alternative for images. This helps screen readers and other assistive technologies understand the content of the image.
+- **Importance of good link text**: You should use descriptive link text to help users understand the purpose of the link. This helps screen readers and other assistive technologies understand the content of the link.
+- **Best practices for making audio and video content accessible**: You should provide captions and transcripts for audio and video content to make it accessible to people with hearing impairments. You should also provide audio descriptions for video content to make it accessible to people with visual impairments.
+- **`tabindex` attribute**: used to specify the order in which elements should receive focus when using the tab key to navigate the website. This can help users with mobility impairments navigate the website more easily.
```

Comment:
I am concerned that the explanation for `tabindex` is giving the wrong impression and encouraging improper use. Basically, you should never use `tabindex` to specify the order that elements should receive focus (in other words, you should never use a `tabindex` value greater than `0`). The order that elements should receive focus should be determined by their position in the DOM (the natural tab order). Messing with that order does not help, but rather hinders  navigation.

The two legit uses for `tabindex`:
 
- Use `tabindex=0` to add an element to the natural tab order. There are primarily two reasons you would need to do this: 1) You are using an element that can't receive keyboard focus by default, such as using a `div` with `role="button"` instead of using a `button` element and 2) there are rare cases where you need to make something focusable that normally wouldn't need to be, for example, enabling keyboard users to scroll overflowing containers because Chrome and Safari still don't allow this by default.
- Add `tabindex="-1" to an element to allow it to receive programmatic focus (this does not place it in the natural tab order). This is used in a variety of situations for "focus management", such as when you need to set the focus on a heading, error message, or dialog, or you are building custom components such as tabbing interfaces or tree components.

The main thing I would stress is that you should never use a `tabindex` value greater than `0` (which leaves just `0` and `-1`) and that the natural tab order should be determined by the elements position in the DOM.